File: exercise/php/day03/src/Preparation/SantaWorkshopService.php
<?php

declare(strict_types = 1);

namespace Preparation;

use InvalidArgumentException;

class SantaWorkshopService
{
	private array $preparedGifts = [];

	public function prepareGift(string $giftName, float $weight, string $color, string $material): Gift
	{
		if (5.0 < $weight) {
			throw new InvalidArgumentException('Gift is too heavy for Santa\'s sleigh');
		}

		$gift = new Gift($giftName, $weight, $color, $material);
		$this->preparedGifts[] = $gift;

		return $gift;
	}
}

File: solution/day03/src/main/java/preparation/SantaWorkshopService.java
package preparation;

import java.util.ArrayList;
import java.util.List;

public class SantaWorkshopService {
    private final List<Gift> preparedGifts = new ArrayList<>();

    public Gift prepareGift(String giftName, double weight, String color, String material) {
        if (weight > 5) {
            throw new IllegalArgumentException("Gift is too heavy for Santa's sleigh");
        }

        var gift = new Gift(giftName, weight, color, material);
        preparedGifts.add(gift);

        return gift;
    }
}